sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/MetadataRequestRetryPolicy.java

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.cosmos.implementation;
+
+import com.azure.cosmos.CosmosException;
+import com.azure.cosmos.implementation.directconnectivity.WebExceptionUtility;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import reactor.core.publisher.Mono;
+
+import java.net.URI;
+
+public class MetadataRequestRetryPolicy implements IRetryPolicy {
+
+    private final static Logger logger = LoggerFactory.getLogger(MetadataRequestRetryPolicy.class);
+    private final GlobalEndpointManager globalEndpointManager;
+    private RxDocumentServiceRequest request;
+
+    public MetadataRequestRetryPolicy(GlobalEndpointManager globalEndpointManager) {
+        this.globalEndpointManager = globalEndpointManager;
+    }
+
+    public void onBeforeSendRequest(RxDocumentServiceRequest request) {
+        this.request = request;
+    }
+
+    private static boolean shouldMarkRegionAsUnavailable(CosmosException exception) {
+
+        // check for network issues or connectivity issues
+        if (WebExceptionUtility.isNetworkFailure(exception)) {
+            return Exceptions.isSubStatusCode(exception, HttpConstants.SubStatusCodes.GATEWAY_ENDPOINT_UNAVAILABLE) ||
+                Exceptions.isSubStatusCode(exception, HttpConstants.SubStatusCodes.GATEWAY_ENDPOINT_READ_TIMEOUT);
```

Comment:
GATEWAY_ENDPOINT_READ_TIMEOUT -> this should not mark the region unavailable. also aligns the logic in ClientRetryPolicy

Reply:
Fixed in next iteration.